var/spack/repos/builtin/packages/sst-core/package.py

```diff
@@ -7,38 +7,72 @@
-    git      = "https://github.com/sstsimulator/sst-core.git"
+    homepage = "https://github.com/sstsimulator"
+    git = "https://github.com/sstsimulator/sst-core.git"
+    url = "https://github.com/sstsimulator/sst-core/releases/download/v9.1.0_Final/sstcore-9.1.0.tar.gz"
 
-    version('develop', branch='devel')
-    version('8.0.0', sha256='34a62425c3209cf80b6bca99cb0dcc328b67fb84ed92d5e6d6c975ad9319ba8a')
+    version('9.1.0', sha256="cfeda39bb2ce9f32032480427517df62e852c0b3713797255e3b838075f3614d")
+    version('devel',   branch='devel')
+    version('master',  branch='master', preferred=True)
 
-    variant('mpi', default=True, description='Support multi-node simulations using MPI')
-    variant('boost', default=False, description='Use boost')
+    variant("pdes-mpi", default=True,
+            description="Build support for parallel discrete event simulation")
+    variant("zoltan",  default=False,
+            description="Use Zoltan for partitioning parallel runs")
+    variant("hdf5",    default=False,
+            description="Build support for HDF5 statistic output")
+    variant("zlib",    default=False,
+            description="Build support for ZLIB compression")
+    variant("preview", default=False,
+            description="Preview build with deprecated features removed")
 
-    depends_on('autoconf@1.68:', type='build', when='@develop')
-    depends_on('automake@1.11.1:', type='build', when='@develop')
-    depends_on('libtool@1.2.4:', type='build', when='@develop')
-    depends_on('m4', type='build', when='@develop')
+    depends_on("python")
+    depends_on("mpi", when="+pdes-mpi")
+    depends_on("zoltan", when="+zoltan")
+    depends_on("hdf5", when="+hdf5")
+    depends_on("zlib", when="+zlib")
 
-    depends_on('python@:2')
-    depends_on('zlib', type='build')
-    depends_on('mpi', when='+mpi')
-    depends_on('boost@1.56.0:', type='build', when='+boost')
+    depends_on('autoconf@1.68:', type='build', when='@devel')
+    depends_on('automake@1.11.1:', type='build', when='@devel')
+    depends_on('libtool@1.2.4:', type='build', when='@devel')
+    depends_on('m4', type='build', when='@devel')
+    depends_on('autoconf@1.68:', type='build', when='@master')
+    depends_on('automake@1.11.1:', type='build', when='@master')
+    depends_on('libtool@1.2.4:', type='build', when='@master')
+    depends_on('m4', type='build', when='@master')
+
+    # force out-of-source builds
+    build_directory = 'spack-build'
+
+    def autoreconf(self, spec, prefix):
+        bash = which('bash')
+        bash('autogen.sh')
 
     def configure_args(self):
         args = []
-        spec = self.spec
+        if "+zoltan" in self.spec:
+            args.append("--with-zoltan=%s" % self.spec["zoltan"].prefix)
+        if "+hdf5" in self.spec:
+            args.append("--with-hdf5=%s" % self.spec["hdf5"].prefix)
+        if "+zlib" in self.spec:
+            args.append("--with-zlib=%s" % self.spec["zlib"].prefix)
```

Comment:
Is it possible to explicitly disable these in an else clause?

Reply:
It is not. These are implicitly off when the `--with-` is omitted. I agree these might make more sense as a `--enable` and `--disable`, but we are stuck with the build system as is-is.  Our users would not be pleased if we changed a bunch of option names.